### PR TITLE
Custodial user and clinic changes

### PR DESCRIPTION
**Pull request by darinkrauss**
See individual commits for details.

@jh-bate

### CODE REVIEW

**Comment by darinkrauss** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;fROzd9mX&#x2F;12-platform-clinic-0-ux-5-dev-2-tst-platform-supports-true-child-accounts"> [platform][clinic] &lt;0:ux 5:dev, 2:tst&gt; Platform supports true child accounts</a>

File: api/signup.go [hunk 1]
```diff
@@ -225,10 +243,64 @@ func (a *Api) acceptSignUp(res http.ResponseWriter, req *http.Request, vars map[
...
+			log.Printf("acceptSignUp: error trying to get user checking email verified [%s]", err.Error())
+			res.WriteHeader(http.StatusInternalServerError)
+			return
+
+		} else if !user.PasswordExists {
+			acceptance := &models.Acceptance{}
+			if req.Body != nil {
+				if err := json.NewDecoder(req.Body).Decode(acceptance); err != nil {
+					log.Printf("acceptSignUp: error decoding acceptance [%#v]", err)
+					a.sendModelAsResWithStatus(res, &status.StatusError{status.NewStatus(http.StatusBadRequest, STATUS_NO_PASSWORD)}, http.StatusBadRequest)
```
Comment (jh-bate) [inline api/signup.go]: why not `a.sendError(` ?

**Comment by jh-bate** [comment]
lgtm @darinkrauss

**Comment by darinkrauss** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;TIEJfsUg&#x2F;5-platform-uploader-clinic-mode-0-ux-3-dev-2-test-enable-clinic-mode-for-the-uploader"> [platform][uploader] [clinic-mode] &lt;0:ux 3:dev 2:test&gt; Enable clinic-mode for the uploader</a>

**Comment by darinkrauss** [comment]
@jh-bate Ping. See just the latest commit for adding an additional error code to the JSON response for email verification failures.

**Comment by darinkrauss** [comment]
@jh-bate Okay, last :two: commits. Had to fix broken tests. 😸

**Comment by darinkrauss** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;HytppRMa&#x2F;16-0-ux-2-dev-0-tst-platform-dependencies-for-blip-combined-verification-password-creation-and-in-clinic-account-creation"> &lt;0:ux 2:dev 0:tst&gt; [platform] dependencies for blip combined verification &amp; password creation and in-clinic account creation</a>

File: api/signup.go [hunk 2]
```diff
@@ -255,25 +264,25 @@ func (a *Api) acceptSignUp(res http.ResponseWriter, req *http.Request, vars map[
 			acceptance := &models.Acceptance{}
 			if req.Body != nil {
 				if err := json.NewDecoder(req.Body).Decode(acceptance); err != nil {
-					a.sendError(res, http.StatusBadRequest, STATUS_NO_PASSWORD, "acceptSignUp: error decoding acceptance: ", err.Error())
+					a.sendErrorWithCode(res, http.StatusConflict, ERROR_NO_PASSWORD, STATUS_NO_PASSWORD, "acceptSignUp: error decoding acceptance: ", err.Error())
```
Comment (jh-bate) [inline api/signup.go]: are they really `409` errors? What is the conflict? - same for other `http.StatusConflict`  status codes below

Comment (darinkrauss) [inline api/signup.go]: It is the only HTTP status code that makes any sort of sense. I suppose we could make the first one above 409 and the rest below 400s.  I think the first one as a 409 make sense as the request (plain email verification) is in conflict with the current state of the resource (it doesn't have a password yet). The rest could be 400 "Bad Syntax".  There was a long conversion about the front-end folks wanting an easier timing figuring out what error actually occurred. So, I added the "code" field, plus changed the HTTP status codes.

@jebeck @krystophv @GordyD Thoughts

Comment (jh-bate) [inline api/signup.go]: yeah I agree with the adding of the code but the change of the status code didn't make sense to me personally. But yes I guess you could argue that 409 might be a better fit for the first error

Comment (darinkrauss) [inline api/signup.go]: @jebeck @krystophv @gniezen @GordyD What do folks think of changing the HTTP response codes below for the missing/invalid password and birthday to 400?  After all, these are technically Bad Requests.  The other two requests (first and last) are really 409 Conflict type errors (no password on account and mismatched birthday).  Thoughts?

Comment (gniezen) [inline api/signup.go]: I really want to say it's a 418, but on a more serious note we should probably use 422:

>The 422 (Unprocessable Entity) status code means the server understands the content type of the request entity (hence a 415(Unsupported Media Type) status code is inappropriate), and the syntax of the request entity is correct (thus a 400 (Bad Request) status code is inappropriate) but was unable to process the contained instructions.  For example, this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions.

Comment (darinkrauss) [inline api/signup.go]: Yea, I could do 422, too. I guess if the JSON is parseable (which it has been at this point) then 400 doesn't really make sense.

Comment (jebeck) [inline api/signup.go]: 422 makes sense to me.

**Pull request merged**